plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroIdentityPlugin.java

```diff
@@ -101,6 +109,36 @@ public TokenManager getTokenManager() {
     }
 
     @Override
+    public UnaryOperator<RestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
+        return AuthcRestHandler::new;
+    }
+
+    class AuthcRestHandler extends DelegatingRestHandler {
+        public AuthcRestHandler(RestHandler original) {
```

Comment:
I see you use the delegating handler here. How does this work exactly? In this case are we redirecting things back to the IdentityPlugin interface to noop? Or what is exactly occurring here?

Reply:
See response here: https://github.com/opensearch-project/OpenSearch/pull/15430#discussion_r1738675922

Its to ensure that any overrides of the defaults within the wrapped RestHandler are delegated to instead of resorting to default implementations from the BaseRestHandler.

Reply:
Removed the DelegatingRestHandler in favor of using the internal RestHandler.Wrapper.